clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -2641,6 +2641,7 @@ scheduler_tmpl_t<RecordType, ReaderType>::pop_from_ready_queue_hold_locks(
             } else {
                 // This input is no longer blocked.
                 res->blocked_time = 0;
+                res->unscheduled = false;
```

Comment:
I'm a little curious why line 2717 didn't take care of this, but agreed it seems reasonable to clear unscheduled here.

Reply:
Line 2717 only applies when we select an input.  We might reject one if it's not past the migration threshold (as the PR description tries to explain).